code/datums/brain_damage/imaginary_friend.dm

```diff
@@ -168,66 +181,299 @@
 		current_image.alpha = 150
 
 	//Add new image to owner and friend
-	if(!hidden && owner.client)
-		owner.client.images |= current_image
+	if(!hidden)
+		add_image_to_clients(current_image, friend_clients)
 
-	client.images |= current_image
+	src.client.images |= current_image
 
 /mob/camera/imaginary_friend/Destroy()
 	if(owner?.client)
 		owner.client.images.Remove(human_image)
 	if(client)
 		client.images.Remove(human_image)
+	owner.imaginary_group -= src
 	return ..()
 
 /mob/camera/imaginary_friend/say(message, bubble_type, list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null, filterproof = null, message_range = 7, datum/saymode/saymode = null)
```

Comment:
This is a terrible way to do it, would it be better to do imaginary friend stuff on send_speech over this??

Reply:
You're right, I spent too much time looking at living say to even see that. I'll restructure the code.